lib/nodefs-handler.js

```diff
@@ -102,7 +114,14 @@ const FsWatchInstances = new Map();
  * @returns {fs.FSWatcher} new fsevents instance
  */
 function createFsWatchInstance(path, options, listener, errHandler, emitRaw) {
+  const originalInode = inode(path)
+
   const handleEvent = (rawEvent, evPath) => {
+    // linux-specific: if inode returns not-null we need to emit inode change event
+    const newInode = inode(path)
```

Comment:
I don't think we can do this. Chokidar is extremely performance-conscious. We count every `stat` call. Consider a 600,000-file dir tree. You're adding two `stat`s, which can degrade performance massively.

Reply:
It should work for individual file watchers only. Directories are not in the path.
I don't see any benchmark suites in the repo though...

Reply:
well, 600k-file tree. I'm using readdirp benchmark https://github.com/paulmillr/readdirp/blob/master/examples/bench.js - we should create something similar here.

Reply:
well, each event does a stat when listener is called [here](https://github.com/paulmillr/chokidar/blob/b67d208fbd7314dd77135adbb84e2b9c509f4345/lib/nodefs-handler.js#L364)
and also there is a stat at the startup time
I think I can reuse these. Which means no new stats will be added (apart from obvious re-stat of an existing path to fix the bug itself)

this readdir bench is just pumping stuff into a directory, my PR will not change that path at all.
it's enabled only when people watch each individual file, which is a pretty bad idea for 600k dir, it will be slow in any case.

anyway, I think I can probably prove that the perf difference is negligible
but it will need some time to get samples of real workloads

Reply:
Nice! Will wait for your results then.

Reply:
@paulmillr okay, now I don't use any new stats. 
it plugs in a completely different place